src/main.ts

```diff
@@ -106,6 +106,20 @@ export function generateFile(typeMap: TypeMap, fileDesc: FileDescriptorProto, pa
   const moduleName = fileDesc.name.replace('.proto', '.ts');
   let file = FileSpec.create(moduleName);
 
+  file = file.addCode(CodeBlock.empty().addStatement('declare var self: any | undefined'));
+  file = file.addCode(CodeBlock.empty().addStatement('declare var window: any | undefined'));
+  // Make sure globalThis is defined
+  let getGlobalThis = FunctionSpec.create('getGlobalThis');
+  ['globalThis', 'self', 'window', 'global'].forEach((candidate) => {
+    getGlobalThis = getGlobalThis
+      .beginControlFlow('if (typeof %L !== "undefined")', candidate)
+      .addStatement('return %L', candidate)
+      .endControlFlow();
+  });
+  getGlobalThis.addStatement('throw new Error("Unable to locate global object")');
```

Comment:
Why is this not added to the output code?

Reply:
Ah whoops, I need to reassign this here.